kubeflow/katib/prototypes/all.jsonnet

```diff
@@ -0,0 +1,23 @@
+// @apiVersion 0.1
+// @name katib
+// @description Kubeflow hyperparameter tuning component
+// @shortDescription hp-tuning
+// @param name string Name to give to each of the components
+// @optionalParam namespace string null Namespace to use for the components. It is automatically inherited from the environment if not set.
+
+local k = import "k.libsonnet";
+
+local vizier = import "kubeflow/katib/vizier.libsonnet";
+local modeldb = import "kubeflow/katib/modeldb.libsonnet";
+local suggestion = import "kubeflow/katib/suggestion.libsonnet";
+
+// updatedParams uses the environment namespace if
+// the namespace parameter is not explicitly set
+local updatedParams = params {
```

Comment:
You do not need this stanza. env should always contain environment because we are now require versions of ksonnet that ensure its true.

Reply:
I don't get it. Do you mean we should always use namespace from env? 
I think the current logic is correct?

Reply:
Correct you can just get namespace from env.
The logic is correct but unnecessary. 

In earlier versions of ksonnet env wasn't available. So the way to get namespace was to make it available via a params.

A later version of ksonnet introduced env which contains namespace. But for backwards compatibility we still wanted to respect namespace in params. But we no longer need to do that.

Reply:
But then all components in one environment will have to have the same namespace?
Shouldn't we allow setting namespace per component?

Reply:
Perhaps in the future. There's an implicit assumption that everything is already in the same namespace. So I wouldn't expect setting different namespaces would just work.

Reply:
I see